### PR TITLE
[exninfo] Deprecate aliases for exception re-raising.

### PR DESCRIPTION
**Pull request by ejgallego**
We make the primitives for backtrace-enriched exceptions canonical in
the `Exninfo` module, deprecating all other aliases.

At some point dependencies between `CErrors` and `Exninfo` were a bit
complex, after recent clean-ups the roles seem much clearer so we can
have a single place for `iraise` and `capture`.

### CODE REVIEW

**Review by Matafou — APPROVED** [review]

**Comment by ejgallego** [comment]
@ppedrot does this look suitable to you?

**Comment by ppedrot** [comment]
To be fair, I don't like this PR. It is touching a lot of files for what I consider to be merely cosmetic purposes. This makes rebases a pain for instance. Furthermore, proxies can have their usefulness, e.g. if we want to squeeze an additional behaviours in `CErrors` in the future.

**Comment by ejgallego** [comment]
> It is touching a lot of files for what I consider to be merely cosmetic purposes.

It removes a duplicated API / alias, IMHO duplicated APIs are quite bad, for example they make grepping for uses quite harder. It is true that refactoring is painful tho; so far even despite the pain we have proceeded to unify aliases. I dunno.

> Furthermore, proxies can have their usefulness, e.g. if we want to squeeze an additional behaviours in `CErrors` in the future.

I'm not sure I see what you mean; so far with the current API exception-related functions are in `Exninfo` as it seems they'll be there for a while, what kind of extra behavior do you see here?

**Comment by ppedrot** [comment]
@ejgallego I won't strongly oppose the PR, but I'd like a few comments from other more impartial devs before merging. Maybe a quick discussion at the weekly call?

**Comment by ejgallego** [comment]
> @ejgallego I won't strongly oppose the PR, but I'd like a few comments from other more impartial devs before merging. Maybe a quick discussion at the weekly call?

Sure, I mean, I can live with the current alias, but the thing is that `CErrors.push` is IMO misleading as that operation has nothing to do with `CErrors`, but with re-raising; indeed it is used in places that don't generate errors but perform control with exceptions, etc....

**Comment by ejgallego** [comment]
Rebased, need asignee.....

**Review by ppedrot — APPROVED** [review]

**Comment by ppedrot** [comment]
Let's merge now, it was already discussed during the last call.

**Comment by herbelin** [comment]
I don't understand well the `exninfo` and I'd prefer that @ppedrot or another maintainer be the assignee.

In passing, I'd like below to tell to locate the error on the first `intro` only but it is located on the whole line.
```
Goal nat.
intro; intros.
^^^^^^^^^^^^^^
```
How should I do? Are several `add_loc` overriding themselves?

**Pull request merged**
